core/src/price.ts

```diff
@@ -45,28 +46,27 @@ export const calculateTransactionGrossProfit = function (auction: Auction): BigN
     if (!auction.marketUnitPrice) {
         return new BigNumber(0);
     }
-    const totalMarketPrice = auction.collateralAmount.multipliedBy(auction.marketUnitPrice);
-    if (totalMarketPrice <= auction.debtDAI) {
-        return totalMarketPrice.minus(auction.totalPrice);
-    }
-    const collateralAmountLimitedByDebt = auction.debtDAI.dividedBy(auction.approximateUnitPrice);
-    const totalMarketPriceLimitedByDebt = collateralAmountLimitedByDebt.multipliedBy(auction.marketUnitPrice);
-    return totalMarketPriceLimitedByDebt.minus(auction.debtDAI);
+    const totalDebtMarketPrice = auction.collateralToCoverDebt.multipliedBy(auction.marketUnitPrice);
+    const totalDebtPrice = auction.collateralToCoverDebt.multipliedBy(auction.approximateUnitPrice);
+    return totalDebtMarketPrice.minus(totalDebtPrice);
 };
 
 export const calculateTransactionCollateralOutcome = function (
     bidAmountDai: BigNumber,
     unitPrice: BigNumber,
-    auction: AuctionTransaction
+    auction: Auction
 ): BigNumber {
     // Based on the clipper contract logic
     // https://github.com/makerdao/dss/blob/60690042965500992490f695cf259256cc94c140/src/clip.sol#L357-L380
     const collateralToBuyForTheBid = bidAmountDai.dividedBy(unitPrice);
     const potentialOutcomeCollateralAmount = BigNumber.minimum(collateralToBuyForTheBid, auction.collateralAmount); // slice
     const potentialOutcomeTotalPrice = potentialOutcomeCollateralAmount.multipliedBy(unitPrice); // owe
+    const approximateDebt = new BigNumber(auction.debtDAI.toPrecision(16, 0));
+    const approximatePotentialOutcomeTotalPrice = new BigNumber(potentialOutcomeTotalPrice.toPrecision(16, 0));
```

Comment:
Please next time use `ROUND_DOWN` variable provided by the bignumber instead of the cryptic number ([which is not round down by the way](https://user-images.githubusercontent.com/3393626/181745611-ed378ec9-3fda-41d2-943b-136a08a37a26.png))

```suggestion
    const approximateDebt = new BigNumber(auction.debtDAI.toPrecision(16, BigNumber.ROUND_DOWN));
    const approximatePotentialOutcomeTotalPrice = new BigNumber(potentialOutcomeTotalPrice.toPrecision(16, BigNumber.ROUND_DOWN));
```